### PR TITLE
Update Baxter for Kinetic

### PR DESCRIPTION
**Pull request by davetcoleman**
Combines https://github.com/ros-planning/moveit_robots/pull/53 and https://github.com/ros-planning/moveit_robots/pull/52 because there was a merge conflict with those, and fixes some more deprecation problems with kinetic

Note Rethink only supports Indigo and will not officially review or support kinetic or this PR, so I think it is fine if another moveit maintainer merges it if needed @IanTheEngineer 

*This PR is sponsored by [Vicarious](http://www.vicarious.com/)*

### CODE REVIEW

**Review by v4hn — APPROVED** [review]
Looks good to me.
CI for this repository seems to be broken though.

Could you have a look before merging please?

File: baxter/baxter_moveit_config/config/baxter_base.srdf.xacro [hunk 1]
```diff
@@ -52,7 +52,9 @@
     </group_state>
     <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
      <!--VIRTUAL JOINT: Purpose: this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
-    <virtual_joint name="world_joint" type="floating" parent_frame="world" child_link="torso" />
+    <virtual_joint name="world_joint" type="floating" parent_frame="world" child_link="base" />
```
Comment (IanTheEngineer) [inline baxter/baxter_moveit_config/config/baxter_base.srdf.xacro:55]: Does this still allow for two arms to be used in the MoveIt GUI? I vaguely remember I had to change it to `torso` in order to enable both arms

File: baxter/baxter_moveit_config/launch/demo_baxter.launch [hunk 2]
```diff
@@ -34,5 +34,10 @@
     <arg name="rviz_config" value="$(arg rviz_config)" />
   </include>
 
-  <include file="$(find baxter_moveit_config)/launch/default_warehouse_db.launch" />
+  <!-- As of ROS Kinetic / Ubuntu 16.04 the mongo driver does not have a binary and must be installed from source.
+       Therefore, this functionality is disabled by default. For more info:
+       https://github.com/ros-planning/moveit/issues/123
```
Comment (IanTheEngineer) [inline baxter/baxter_moveit_config/launch/demo_baxter.launch:39]: +1 very helpful comment

**Review by IanTheEngineer — COMMENTED** [review]
Provided that `base` still functions as the `child_link` inside the `virtual_joint`, these changes look good.

**Comment by IanTheEngineer** [comment]
The `ci` seems to output a funny message about a non-existent package:
[`[build] Given package '--no-deps atlas_moveit_config
atlas_v3_moveit_config
baxter_ikfast_left_arm_plugin
baxter_ikfast_right_arm_plugin
baxter_moveit_config
iri_wam_moveit_config
moveit_robots
r2_moveit_generated' is not in the workspace`](https://travis-ci.org/ros-planning/moveit_robots/jobs/215061861#L523-L530)

File: baxter/baxter_moveit_config/config/baxter_base.srdf.xacro (hunk 1 above)
Comment (davetcoleman) [reply baxter/baxter_moveit_config/config/baxter_base.srdf.xacro:55]: The virtual joint should connect to the root of the URDF tree. I tested this yesterday and was able to plan with both arms repeatedly.

**Comment by IanTheEngineer** [comment]
I'm fine with merging these changes. @v4hn & @130s, should we be bothered at all with the `ci-build` failing? I don't see how Dave's changes could have broken `ci`.

File: baxter/baxter_moveit_config/config/baxter_base.srdf.xacro (hunk 1 above)
Comment (IanTheEngineer) [reply baxter/baxter_moveit_config/config/baxter_base.srdf.xacro:55]: I'll give this test a whirl as well. Might either have been an issue with end effectors or possibly this was a bug that was fixed in recent updates to MoveIt... Regardless, if it works, I'm fine moving to the `base`  frame.

**Comment by IanTheEngineer** [comment]
I tested your branch out on my laptop and get two errors, one warning. None of them seem to affect performance:
```
ERROR: cannot launch node of type [warehouse_ros/mongo_wrapper_ros.py]: can't locate node [mongo_wrapper_ros.py] in package [warehouse_ros]
[ INFO] [1490623607.735298317]: Loading robot model 'baxter'...
[ WARN] [1490623607.735562021]: Skipping virtual joint 'world_joint' because its child frame 'base' does not match the URDF frame 'world'
[ERROR] [1490623635.647106440, 365.237000000]: Unknown marker name: 'EE:goal_right_gripper' (not published by RobotInteraction class)
```
The WARN does seem to publish repeatedly.

**Comment by v4hn** [comment]
> should we be bothered at all with the ci-build failing? I don't see how Dave's changes could have broken ci.

The thing is, @davetcoleman is the maintainer who insists most on successful CI tests. :)
So I'll definitely leave it to him to look at it and answer that question.

I'm fine with it either way.

**Comment by 130s** [comment]
> The `ci` seems to output a funny message about a non-existent package:
> [`[build] Given package '--no-deps atlas_moveit_config
> atlas_v3_moveit_config
> baxter_ikfast_left_arm_plugin
> baxter_ikfast_right_arm_plugin
> baxter_moveit_config
> iri_wam_moveit_config
> moveit_robots
> r2_moveit_generated' is not in the workspace`](https://travis-ci.org/ros-planning/moveit_robots/jobs/215061861#L523-L530)

This is weird. I tried to reproduce locally but this didn't occur and the corresponding [catkin build](https://github.com/ros-planning/moveit_ci/blob/d11d7d0b45cebf77430adc29b05e36e9db3dee22/travis.sh#L161) command passed without any error. I re-started Travis to see if the same happens.
(If the job ran on Sunday night west US time, I saw Travis/GitHub behaved strange).

**Comment by davetcoleman** [comment]
I worked on fixing the CI last week but its having a weird error on running the tests that do not show any errors. I haven't been able to figure out why yet, even after running the CI locally...

**Comment by davetcoleman** [comment]
I have been working on fixing the CI for several days now and will address the issue (seems to me the line breaks are the problem) but I need to merge this issue so I can make an announcement about Baxter/Gazebo working.

**Pull request merged**